### PR TITLE
AStar Whitelist Procs

### PR DESCRIPTION
**Pull request by TheDarkElites**
## About the Pull Request

Adds in the required procs to be able to use AStar with a whitelist.

## Why It's Good For The Game

We're gonna have AI that needs to be able to find paths through normally dense objects, as some SCPs will inevitably be able to pass through random things due to their anomalous nature. Right now, this is needed for 173's AI to allow him to pathfind through windows and doors.

## Changelog

:cl:
code: Astar Whitelist Tools
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
I need this to continue reworking 173's AI so like pretty please review and merge soon.

**Comment by TheDarkElites** [comment]
currently test merged, thank you :)

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -250,6 +250,16 @@ var/const/enterloopsanity = 100
 		if(T.x == src.x || T.y == src.y)
 			. += T
 
+/turf/proc/AdjacentTurfsWithWhitelist(var/list/whitelist, check_blockage = TRUE)
```
Comment (EgorDinamit) [inline code/game/turfs/turf.dm]:
```suggestion
/turf/proc/AdjacentTurfsWithWhitelist(check_blockage = TRUE, list/whitelist)
```

File: code/_helpers/unsorted.dm [hunk 2]
```diff
@@ -178,31 +178,25 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
 
-	if(DirBlocked(A,adir)) return 1
-	if(DirBlocked(B,rdir)) return 1
+	if(DirBlocked(A,adir, whitelist)) return 1
+	if(DirBlocked(B,rdir, whitelist)) return 1
 	return 0
 
 
-/proc/DirBlocked(turf/loc,dir)
+/proc/DirBlocked(turf/loc, dir, var/list/whitelist = list())
```
Comment (Tsurupeta) [inline code/_helpers/unsorted.dm]:
```suggestion
/proc/DirBlocked(turf/loc, dir, list/whitelist = list())
```

File: code/_helpers/unsorted.dm [hunk 3]
```diff
@@ -166,7 +166,7 @@ Turf and target are seperate in case you want to teleport some distance from a t
 
 
 
-/proc/LinkBlocked(turf/A, turf/B)
+/proc/LinkBlocked(turf/A, turf/B, var/list/whitelist)
```
Comment (Tsurupeta) [inline code/_helpers/unsorted.dm]:
```suggestion
/proc/LinkBlocked(turf/A, turf/B, list/whitelist)
```

**Review by EgorDinamit — APPROVED** [review]

**Pull request merged**